src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1855,17 +2033,89 @@ bool isRefEscape
             ImmutableArray<RefKind> argRefKindsOpt,
             ImmutableArray<int> argsToParamsOpt,
             bool isRefEscape,
-            bool checkingMethodArgumentsMustMatch,
             bool ignoreArglistRefKinds,
-            ArrayBuilder<(ParameterSymbol? Parameter, BoundExpression Argument, bool IsRefEscape)> result)
+            ArrayBuilder<EscapeValue> escapeValues)
+        {
+            GetFilteredArguments(
+                symbol,
+                receiver,
+                parameters,
+                argsOpt,
+                argRefKindsOpt,
+                argsToParamsOpt,
+                ignoreArglistRefKinds,
+                mixableArguments: null,
+                escapeValues);
+
+            bool includeRefScopes = shouldIncludeRefScopes();
+            int index = 0;
+            while (index < escapeValues.Count)
+            {
+                var tuple = escapeValues[index];
+                if (!includeRefScopes && tuple.IsRefEscape)
+                {
+                    escapeValues.RemoveAt(index);
+                }
+                else
+                {
+                    index++;
+                }
+            }
+
+            bool shouldIncludeRefScopes()
```

Comment:
>bool [](http://example.com/codeflow?start=12&length=4)

Consider declaring this a `static` local function and passing `symbol` and `isRefEscape` explicitly as parameters, so it's clear what state the return value depends on.